modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/disk/PlainExternalResult.java

```diff
@@ -115,16 +115,19 @@ private void spillRows() {
         if (F.isEmpty(rowBuff))
             return;
 
-        data.store(rowBuff);
+        long swapped = data.store(rowBuff);
+
+        memTracker.swap(swapped);
+        this.swapped += swapped;
```

Comment:
In my opinion it is not the best way to track data amount that spilled to disk. I can see two drawbacks here:
- Code in external results became more complicated because of the tracking part
- It is easy very easy to make a mistake and forget to count some disk offloads with this approach. Like it happened for `ExternalResultHashIndex`

I think that using specialized `FileIoFactory` for tracking spilled files is more solid solution here. See example `TrackableFileIoFactory` in gg-27202 branch.